core-runner/src/runner.rs

```diff
@@ -210,6 +210,64 @@ pub struct ProgramInitialization {
     pub code: Vec<u8>,
 }
 
+/// New message dispatch request.
+///
+/// Message is dispatched from some identity to some identity, both should be known in advance.
+pub struct MessageDispatch {
+    /// Identity of the message origin.
+    pub source_id: ProgramId,
+    /// Identity of the destination.
+    pub destination_id: ProgramId,
+    /// Message payload and economic data.
+    pub data: ExtMessage,
+}
+
+impl MessageDispatch {
```

Comment:
Why to not just implement `From<MessageDispatch> for Message` and then use `from()` and `into()`?

Reply:
because it is in the different crate?

Reply:
`Message` is local there, not so for `gear-core-runner`